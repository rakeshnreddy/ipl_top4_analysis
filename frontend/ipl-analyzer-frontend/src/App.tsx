import React from 'react';
import './App.css';
import CurrentStandings from './components/CurrentStandings';
import OverallProbabilities from './components/OverallProbabilities';
import DetailedTeamAnalysis from './components/DetailedTeamAnalysis';
import QualificationPath from './components/QualificationPath';
import ScenarioSimulation from './components/ScenarioSimulation';

const App: React.FC = () => {
  console.log('App.tsx - BASE_URL:', import.meta.env.BASE_URL);
  return (
    <div id="app-container">
      <header>
        <h1>IPL Qualification Analyzer</h1>
      </header>

      <section id="current-standings" aria-labelledby="current-standings-heading">
        <h2 id="current-standings-heading">Current Standings</h2>
        <CurrentStandings />
      </section>

      <section id="overall-probabilities" aria-labelledby="overall-probabilities-heading">
        <h2 id="overall-probabilities-heading">Overall Qualification Probabilities</h2>
        <OverallProbabilities />
      </section>

      <section id="team-analysis" aria-labelledby="team-analysis-heading">
        <h2 id="team-analysis-heading">Detailed Team Analysis</h2>
        <DetailedTeamAnalysis />
      </section>

      <section id="qualification-path" aria-labelledby="qualification-path-heading">
        <h2 id="qualification-path-heading">Qualification Path</h2>
        <QualificationPath />
      </section>

      <section id="scenario-simulation" aria-labelledby="scenario-simulation-heading">
        <h2 id="scenario-simulation-heading">Simulate One Scenario</h2>
        <ScenarioSimulation />
      </section>

      <footer>
        <p>&copy; {new Date().getFullYear()} IPL Analyzer. All rights reserved.</p>
        <p>Data is for informational purposes only.</p>
      </footer>
    </div>
  );
};

export default App;
