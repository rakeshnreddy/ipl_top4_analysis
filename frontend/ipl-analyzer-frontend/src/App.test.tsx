/// <reference types="vitest/globals" />
import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import App from './App'; // Assuming App.tsx is in the same directory or path is adjusted

// Mock child components to simplify App testing and focus on App's own structure/styles
vi.mock('./components/CurrentStandings', () => ({
  default: () => <div data-testid="current-standings-mock">CurrentStandings Mock</div>,
}));
vi.mock('./components/OverallProbabilities', () => ({
  default: () => <div data-testid="overall-probabilities-mock">OverallProbabilities Mock</div>,
}));
vi.mock('./components/DetailedTeamAnalysis', () => ({
  default: () => <div data-testid="detailed-team-analysis-mock">DetailedTeamAnalysis Mock</div>,
}));
vi.mock('./components/QualificationPath', () => ({
  default: () => <div data-testid="qualification-path-mock">QualificationPath Mock</div>,
}));
vi.mock('./components/ScenarioSimulation', () => ({
  default: () => <div data-testid="scenario-simulation-mock">ScenarioSimulation Mock</div>,
}));

describe('App Component', () => {
  it('Test 1: Renders correctly and matches snapshot', () => {
    const { container } = render(<App />);
    
    // Check for main sections by aria-labelledby or heading text
    expect(screen.getByRole('heading', { name: /IPL Qualification Analyzer/i })).toBeInTheDocument();
    expect(screen.getByRole('heading', { name: /Current Standings/i })).toBeInTheDocument();
    expect(screen.getByTestId('current-standings-mock')).toBeInTheDocument();
    expect(screen.getByRole('heading', { name: /Overall Qualification Probabilities/i })).toBeInTheDocument();
    expect(screen.getByTestId('overall-probabilities-mock')).toBeInTheDocument();
    expect(screen.getByRole('heading', { name: /Detailed Team Analysis/i })).toBeInTheDocument();
    expect(screen.getByTestId('detailed-team-analysis-mock')).toBeInTheDocument();
    expect(screen.getByRole('heading', { name: /Qualification Path/i })).toBeInTheDocument();
    expect(screen.getByTestId('qualification-path-mock')).toBeInTheDocument();
    expect(screen.getByRole('heading', { name: /Simulate One Scenario/i })).toBeInTheDocument();
    expect(screen.getByTestId('scenario-simulation-mock')).toBeInTheDocument();
    
    // Snapshot test
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Test 2: Footer renders correctly', () => {
    render(<App />);
    const year = new Date().getFullYear();
    expect(screen.getByText(`Â© ${year} IPL Analyzer. All rights reserved.`)).toBeInTheDocument();
    expect(screen.getByText('Data is for informational purposes only.')).toBeInTheDocument();
  });

  // Add more tests here if there's specific logic in App.tsx related to themes or styles
  // For example, if App.tsx had a theme toggler:
  // it('Test 3: Applies dark mode class when dark mode is active', () => {
  //   // Mock or set up a condition for dark mode
  //   // render(<App />);
  //   // expect(screen.getByTestId('app-container')).toHaveClass('dark-theme'); 
  //   // This depends on how dark mode is implemented.
  // });
});
