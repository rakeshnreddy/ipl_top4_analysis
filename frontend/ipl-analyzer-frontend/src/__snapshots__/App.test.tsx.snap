// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App Component > Test 1: Renders correctly and matches snapshot 1`] = `
<div
  id="app-container"
>
  <header>
    <h1>
      IPL Qualification Analyzer
    </h1>
    <div
      aria-label="Theme selection"
      class="_toggleContainer_f72571"
      role="radiogroup"
    >
      <button
        aria-checked="false"
        aria-label="Select Light theme"
        class="_toggleButton_f72571 "
        role="radio"
      >
        Light
      </button>
      <button
        aria-checked="false"
        aria-label="Select Dark theme"
        class="_toggleButton_f72571 "
        role="radio"
      >
        Dark
      </button>
      <button
        aria-checked="true"
        aria-label="Select System theme"
        class="_toggleButton_f72571 _active_f72571"
        role="radio"
      >
        System
      </button>
    </div>
  </header>
  <section
    aria-labelledby="current-standings-heading"
    id="current-standings"
  >
    <div
      class="_card_6217ab"
    >
      <h2
        id="current-standings-heading"
      >
        Current Standings
      </h2>
      <div
        data-testid="current-standings-mock"
      >
        CurrentStandings Mock
      </div>
    </div>
  </section>
  <section
    aria-labelledby="overall-probabilities-heading"
    id="overall-probabilities"
  >
    <div
      class="_card_6217ab"
    >
      <h2
        id="overall-probabilities-heading"
      >
        Overall Qualification Probabilities
      </h2>
      <div
        data-testid="overall-probabilities-mock"
      >
        OverallProbabilities Mock
      </div>
    </div>
  </section>
  <section
    aria-labelledby="team-analysis-heading"
    id="team-analysis"
  >
    <div
      class="_card_6217ab"
    >
      <h2
        id="team-analysis-heading"
      >
        Detailed Team Analysis
      </h2>
      <div
        data-testid="detailed-team-analysis-mock"
      >
        DetailedTeamAnalysis Mock
      </div>
    </div>
  </section>
  <section
    aria-labelledby="qualification-path-heading"
    id="qualification-path"
  >
    <div
      class="_card_6217ab"
    >
      <h2
        id="qualification-path-heading"
      >
        Qualification Path
      </h2>
      <div
        data-testid="qualification-path-mock"
      >
        QualificationPath Mock
      </div>
    </div>
  </section>
  <section
    aria-labelledby="scenario-simulation-heading"
    id="scenario-simulation"
  >
    <div
      class="_card_6217ab"
    >
      <h2
        id="scenario-simulation-heading"
      >
        Simulate One Scenario
      </h2>
      <div
        data-testid="scenario-simulation-mock"
      >
        ScenarioSimulation Mock
      </div>
    </div>
  </section>
  <footer>
    <p>
      Â© 
      2025
       IPL Analyzer. All rights reserved.
    </p>
    <p>
      Data is for informational purposes only.
    </p>
  </footer>
</div>
`;
